File: BUEN_SABOR_BACKEND/src/main/java/com/lab4/buen_sabor_backend/model/DetallePedido.java
package com.lab4.buen_sabor_backend.model;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DetallePedido extends Master{

    private Integer cantidad;
    private Double subTotal;

    @ManyToOne
    @JoinColumn(name = "pedido_id")
    @JsonIgnore
    private Pedido pedido;

    @ManyToOne
    @JoinColumn(name = "articulo_id")
    private Articulo articulo;
}

File: BUEN_SABOR_FRONTEND/src/models/DetallePedido.ts
import Articulo from "./Articulo";
import type Promocion from "./Promocion";

export default class DetallePedido {
    id?: number;
    cantidad: number = 0;
    subTotal: number = 0;
    articulo?: Articulo;
    promocion?: Promocion
    eliminado!: boolean;

}